code/bookmarklet_builder.js

```diff
@@ -0,0 +1,94 @@
+(function() {
+  // Live version
+  var HOST = 'https://r.keymanweb.com';
+
+  // // Debug / dev version
+  // var HOST = 'http://r.keymanweb.com.local';
+
+  function construct_bookmarklet_stylesheet() {
+    var style = document.createElement('style');
+
+    var imgHref = `${HOST}/code/keymanweb-icon-16.png`;
+
+    var styleSpec = `
+    /* Easier to override when less specific */
+    .keyman-bookmarklet {
+      font-size: 11px;
+    }
+
+    .keyman-bookmarklet a {
+      padding: 4px 4px 4px 24px;
+      background: #D3DAED url('${imgHref}') 4px 4px no-repeat scroll;
+      box-shadow: 3px 3px 6px rgba(50,52,56,0.3);
+      text-decoration: none;
+      color: #000000;
+      text-align: left;
+      display: inline-block;
+      height: 16px;
+      overflow: hidden;
+      white-space: nowrap;
+    }
+    `;
+
+    style.id='keyman-bookmarklet-style';
+    style.appendChild(document.createTextNode(styleSpec));
+
+    return style;
+  }
+
+  var bml_style = {
+    element: construct_bookmarklet_stylesheet(),
+    isLoaded: false
+  };
+
+  function construct_bookmarklet_code(kbdid, langid) {
+    var code = `
+(function(e) {
+  e=document.createElement('script');
+  e.type='text/javascript';
+  e.onload=() => {
+    loadKeymanWebBookmarklet('${kbdid}', '${langid}');
+  };
+  e.src='${HOST}/code/bookmarklet_loader.js';
+  document.body.appendChild(e);
+})()`;
```

Comment:
```suggestion
  function construct_bookmarklet_code(host, kbdid, langid) {
    var code = `
(function(e) {
  e=document.createElement('script');
  e.type='text/javascript';
  e.onload=() => {
    loadKeymanWebBookmarklet('${kbdid}', '${langid}');
  };
  e.src='${host}/code/bookmarklet_loader.js';
  document.body.appendChild(e);
})()`;
```

If we move the css into its own file and link to it directly from the parent site, and pass the hostname we want to use into this construct function, then we no longer need to hard code a site (`HOST`) self-reference into this script, which makes testing on local and staging much easier, as well as meaning that we are not so tied to r.keymanweb.com in the future.

Reply:
So, that'd be here _and_ in `construct_bookmarklet`, with the sites utilizing the script passing in the value for `host`?  Making sure I understand what you're saying correctly.

Reply:
Yeah, that's right :grin:

Reply:
I'm having trouble following through on that for keymanweb.com; its calling code is within a raw JS file, not anything built on server via PHP.  As a result, it's not clear how to configure that JS to pass the appropriate server-var for the proper r.keymanweb.com version to the method signature you're requesting.  (Refer to https://github.com/keymanapp/keymanweb.com/blob/a0a4e91809352a78f58103abdefcc6352db2674d/cdn/dev/js/kmwlive.js#L451 - that's the line that would need the server var that only exists in PHP-land.)

Or is the "right answer" to turn that JS into PHP?  I mean, it's possible... but that feels a bit out of scope for this PR.  Or am I missing an easier-to-implement approach here?

The keyman.com site's use is in PHP, so that one is indeed easy to tweak according with your request.

Reply:
> Or am I missing an easier-to-implement approach here?

There is a cute shortcut we can use. The variable `resourceDomain` is set to `'https://r.keymanweb.com'` (or similar, per environment) in the page head, before kmwlive.js is loaded:

https://github.com/keymanapp/keymanweb.com/blob/4f841ef0be0d00b93ba6a7e20cc288b0872a17fa/inc/head.php#L101